cylc/uiserver/tests/test_handlers.py

```diff
@@ -44,18 +47,35 @@ def get_app(self) -> Application:
             ]
         )
 
+    @pytest.mark.usefixtures("mock_authentication")
     def test_jupyterhub_version_returned(self):
-        with patch.object(MainHandler, 'get_current_user') as mocked:
-            mocked.return_value = {
-                'name': 'yossarian',
-                'server': 'catch:22'
-            }
-            with open(os.path.join(self.tempdir, "index.html"), "w+") as nf:
-                nf.write("TESTING!")
-                nf.flush()
-                response = self.fetch("/")  # type: HTTPResponse
-                assert response.body == b"TESTING!"
-                assert "X-JupyterHub-Version" in response.headers
+        with open(os.path.join(self.tempdir, "index.html"), "w+") as nf:
+            nf.write("TESTING!")
+            nf.flush()
+            response = self.fetch("/")  # type: HTTPResponse
+            assert response.body == b"TESTING!"
+            assert "X-JupyterHub-Version" in response.headers
+
+    @pytest.mark.usefixtures("mock_authentication")
+    def test_authorised_and_authenticated(self):
+        """Test 500 HTTP response (because index.html missing)."""
+        response = self.fetch('/')
+        assert response.code == 500
+
+    # TODO: can't get this test to run due to handler setup?
+    # @pytest.mark.usefixtures("mock_authentication_none")
+    # def test_unauthenticated(self) -> None:
+    #     """Test 403 HTTP response (unauthenticated)."""
+    #     response = self.fetch('/')
+    #     assert response.code == 403
+    #     # assert response.reason == 'Forbidden'
```

Comment:
I get 500 errors for these tests as the UIS hits an internal error before authentication has the change to fail.

Should find a way to configure things so these tests run in the future, in the mean time can test manually (behaviour unchanged).

Reply:
https://github.com/cylc/cylc-uiserver/issues/new?assignees=&labels=&template=feature-request.md&title=